### PR TITLE
refactor: remove job rights

### PR DESCRIPTION
**Pull request by SnickerDoodlePop**
>remove class definitions for `JobRights` and `JobRightsDomain`
>replace occurences of `JobRights` with `dict`
>update snapshots to reflect absent job rights

### CODE REVIEW

**Comment by SnickerDoodlePop** [comment]
Deepsource is complaining about an async method being overridden as non-async; I am ignoring this.

File: virtool/http/client.py [hunk 1]
```diff
@@ -63,7 +61,7 @@ def has_permission(self, permission: str) -> bool:
 
 
 class JobClient(AbstractClient):
-    def __init__(self, job_id, rights: JobRights):
+    def __init__(self, job_id, rights: dict):
```
Comment (igboyes) [inline virtool/http/client.py:64]: Is `rights` used for anything in `JobClient` now? It should be removed if not.

File: virtool/jobs/data.py [hunk 2]
```diff
@@ -168,7 +163,7 @@ async def create(
         workflow: str,
         job_args: Document,
         user_id: str,
-        rights: JobRights,
+        rights: dict,
```
Comment (igboyes) [inline virtool/jobs/data.py:166]: Remove argument. It is always going to be an empty dict.

File: virtool/jobs/db.py [hunk 3]
```diff
@@ -186,7 +186,7 @@ async def create_job(
         "workflow": workflow,
         "args": job_args,
         "key": None,
-        "rights": rights.as_dict(),
+        "rights": rights,
```
Comment (igboyes) [inline virtool/jobs/db.py:189]: Remove argument. It is never going to be a non-empty dict.
```suggestion
        "rights": {},
```

**Review by igboyes — CHANGES_REQUESTED** [review]

**Review by igboyes — APPROVED** [review]

**Pull request merged**